include/benchmark/reporter.h

```diff
@@ -90,33 +111,69 @@ class ConsoleReporter : public BenchmarkReporter {
  public:
   virtual bool ReportContext(const Context& context);
   virtual void ReportRuns(const std::vector<Run>& reports);
-protected:
+
+ protected:
   virtual void PrintRunData(const Run& report);
 
   size_t name_field_width_;
 };
 
 class JSONReporter : public BenchmarkReporter {
-public:
+ public:
   JSONReporter() : first_report_(true) {}
   virtual bool ReportContext(const Context& context);
   virtual void ReportRuns(const std::vector<Run>& reports);
   virtual void Finalize();
 
-private:
+ private:
   void PrintRunData(const Run& report);
 
   bool first_report_;
 };
 
 class CSVReporter : public BenchmarkReporter {
-public:
+ public:
   virtual bool ReportContext(const Context& context);
   virtual void ReportRuns(const std::vector<Run>& reports);
 
-private:
+ private:
   void PrintRunData(const Run& report);
 };
 
-} // end namespace benchmark
-#endif // BENCHMARK_REPORTER_H_
+class HTMLReporter : public BenchmarkReporter {
+ public:
+  virtual bool ReportContext(const Context& context);
+  virtual void ReportRuns(const std::vector<Run>& reports);
+  virtual void Finalize();
+
+  struct RunData {
+    int64_t iterations;
+    double realTime;
+    double cpuTime;
+
+    double bytesSecond;
+    double itemsSecond;
+    int range_x;
+  };
+
+  struct BenchmarkData {
+    std::string name;
+    std::vector<RunData> runData;
+  };
+
+ private:
+  void writeFile(const char* file) const;
+  std::string replaceHTMLSpecialChars(const std::string& label) const;
+
+  void printHTML(std::ostream& out, const std::string& html) const;
+
+  void appendRunDataTo(std::vector<BenchmarkData> *container, const Run &data, bool isStddev) const;
+
+  std::vector<BenchmarkData> benchmarkTests_Line;
```

Comment:
no underscores in variable names please. 

https://google.github.io/styleguide/cppguide.html